Sources/XcodeProj/Project/XcodeProj.swift

```diff
@@ -133,74 +148,122 @@ extension XcodeProj: Writable {
     ///
-    /// - Parameter path: path to `.xcodeproj` file.
-    /// - Parameter override: if project should be overridden. Default is true.
-    ///   If true will remove all existing schemes before writing.
-    ///   If false will throw error if scheme already exists at the given path.
-    public func writeSchemes(path: Path, override: Bool = true) throws {
-        guard let sharedData = sharedData else { return }
-
-        let schemesPath = XcodeProj.schemesPath(path)
-        if override, schemesPath.exists {
-            try schemesPath.delete()
-        }
-        try schemesPath.mkpath()
-        for scheme in sharedData.schemes {
-            try scheme.write(path: XcodeProj.schemePath(path, schemeName: scheme.name), override: override)
-        }
+    /// - Parameter path: parent folder of schemes folder (xcshareddata or xcuserdata)
+    /// - Returns: scheme management plist path relative to the given path.
+    public static func schemeManagementPath(_ path: Path) -> Path {
+        XcodeProj.schemesPath(path) + "xcschememanagement.plist"
     }
 
     /// Returns debugger folder path relative to the given path.
     ///
-    /// - Parameter path: `.xcodeproj` file path
-    /// - Parameter schemeName: scheme name
+    /// - Parameter path: parent folder of debugger folder (xcshareddata or xcuserdata)
     /// - Returns: debugger folder path relative to the given path.
     public static func debuggerPath(_ path: Path) -> Path {
-        XcodeProj.sharedDataPath(path) + "xcdebugger"
+        path + "xcdebugger"
     }
 
     /// Returns breakpoints plist path relative to the given path.
     ///
-    /// - Parameter path: `.xcodeproj` file path
-    /// - Parameter schemeName: scheme name
+    /// - Parameter path: parent folder of debugger folder (xcshareddata or xcuserdata)
     /// - Returns: breakpoints plist path relative to the given path.
     public static func breakPointsPath(_ path: Path) -> Path {
         XcodeProj.debuggerPath(path) + "Breakpoints_v2.xcbkptlist"
     }
 
-    /// Writes all project breakpoints to the given path.
+    /// Writes all schemes to the given path.
     ///
-    /// - Parameter path: path to `.xcodeproj` file.
+    /// - Parameter path: parent folder of schemes folder (xcshareddata or xcuserdata)
+    /// - Parameter override: if schemes should be overridden. Default is true.
+    ///   If true will remove all existing schemes before writing.
+    ///   If false will throw error if scheme already exists at the given path.
+    public static func writeSchemes(schemes: [XCScheme], path: Path, override: Bool = true) throws {
+        let schemesPath = XcodeProj.schemesPath(path)
+        if override, schemesPath.exists {
+            try schemesPath.delete()
+        }
+
+        try schemesPath.mkpath()
+        for scheme in schemes {
+            try scheme.write(path: XcodeProj.schemePath(path, schemeName: scheme.name), override: override)
+        }
+    }
```

Comment:
This is a static func on XcodeProj because it needs to be used both by `XCSharedData` and `XCUserData`. 

An extension on Array would also be possible if you prefer that. Something like: 

extension Array: Writable where Element == XCScheme {
    public func write(path: Path, override: Bool) throws {
        if override, path.exists {
            try path.delete()
        }

        try path.mkpath()
        for scheme in self {
            try scheme.write(path: path + "\(scheme.name).xcscheme",
                             override: override)
        }
    }
}